src/main/java/net/sf/jabref/importer/fileformat/MedlineImporter.java

```diff
@@ -83,50 +118,349 @@ public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
     public ParserResult importDatabase(BufferedReader reader) throws IOException {
         Objects.requireNonNull(reader);
 
-        // Obtain a factory object for creating SAX parsers
-        SAXParserFactory parserFactory = SAXParserFactory.newInstance();
-
-        // Configure the factory object to specify attributes of the parsers it
-        // creates
-        parserFactory.setValidating(true);
-        parserFactory.setNamespaceAware(true);
-
-        // Now create a SAXParser object
         List<BibEntry> bibItems = new ArrayList<>();
         try {
-            SAXParser parser = parserFactory.newSAXParser(); // May throw
-            // exceptions
-            MedlineHandler handler = new MedlineHandler();
-            // Start the parser. It reads the file and calls methods of the
-            // handler.
-            parser.parse(new InputSource(reader), handler);
-
-            // Switch this to true if you want to make a local copy for testing.
-            if (false) {
-                reader.reset();
-                try (FileOutputStream out = new FileOutputStream(new File("/home/alver/ut.txt"))) {
-                    int c;
-                    while ((c = reader.read()) != -1) {
-                        out.write((char) c);
+            JAXBContext context = JAXBContext.newInstance("net.sf.jabref.logic.fileformats.medline");
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+            Object o = unmarshaller.unmarshal(reader);
+
+            //check whether we have an article set or just an article
+            if (o instanceof PubmedArticleSet) {
+                PubmedArticleSet articleSet = (PubmedArticleSet) o;
+                for (int i = 0; i < articleSet.getPubmedArticleOrPubmedBookArticle().size(); i++) {
+                    PubmedArticle currentArticle = (PubmedArticle) articleSet.getPubmedArticleOrPubmedBookArticle()
+                            .get(i);
+                    parseArticle(currentArticle, bibItems);
+                }
+            } else {
+                PubmedArticle article = (PubmedArticle) o;
+                parseArticle(article, bibItems);
+            }
+        } catch (JAXBException e1) {
+            LOGGER.warn("Could not parse document");
```

Comment:
Don't get the importer an OutputPrinter passed? That should be used instead
of directly communicating with the UI. Reason: Usage of JabRef in the
command line without UI